packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -976,7 +980,10 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             message: error?.message,
         });
 
-        this.summaryManager.dispose();
+        if (this.summaryManager !== undefined) {
+            this.summaryManager.off("summarizerWarning", this.raiseContainerWarning);
```

Comment:
Sort of a question out of curiosity: Would it be simpler to just remove all listeners in `SummaryManager.dispose()`?

I was thinking about this in general, and I know that from this code's perspective it doesn't necessarily know whether SummaryManager will remove its listeners or not, and since this code attached the listener, it sort of "owns" that, and so it has the responsibility to clean up. But I was thinking we could make a rule: that disposable event listeners call removeAllListeners on dispose to reduce excess code?

Reply:
I'm anti-removeAllListeners in general.  It's not detectable to the listeners, and correctly handling the disposal of an object you're listening to likely requires more than just detaching the listener.  It also goes against the core benefit of the emitter pattern that the emitter gets to remain oblivious to whoever might be listening.

Main threat is just a memory leak which can be debugged reasonably easily (and in the disposal case probably doesn't matter that much since you'll probably be hitting F5 soon anyway).

Reply:
Yeah like I get it in the case that the emitter isn't disposed, but the listening object is. Makes sense to me, but maybe it would be nice to have a helper class or something that groups them so you can just call remove once, and not have to specify the event names again.

Anyway, makes sense to me! Resolved.